lib/ansible/modules/files/flatpak.py

```diff
@@ -0,0 +1,249 @@
+  type: string
+  sample: "/usr/bin/flatpak install --user -y flathub org.gnome.Calculator"
+msg:
+  description: Module error message
+  returned: failure
+  type: string
+  sample: "Executable '/usr/local/bin/flatpak' was not found on the system."
+rc:
+  description: Return code from flatpak binary
+  returned: When a flatpak command has been executed
+  type: int
+  sample: 0
+stderr:
+  description: Error output from flatpak binary
+  returned: When a flatpak command has been executed
+  type: string
+  sample: "error: Error searching remote flathub: Can't find ref org.gnome.KDE"
+stdout:
+  description: Output from flatpak binary
+  returned: When a flatpak command has been executed
+  type: string
+  sample: "org.gnome.Calendar/x86_64/stable\tcurrent\norg.gnome.gitg/x86_64/stable\tcurrent\n"
+'''
+
+import subprocess
+from ansible.module_utils.six.moves.urllib.parse import urlparse
+from ansible.module_utils.basic import AnsibleModule
+
+
+def install_flat(module, binary, remote, name, method):
+    """Add a new flatpak."""
+    global result
+    if 'http://' in name or 'https://' in name:
+        command = "{0} install --{1} -y {2}".format(binary, method, name)
+    else:
+        command = "{0} install --{1} -y {2} {3}".format(binary, method, remote, name)
+    _flatpak_command(module, module.check_mode, command)
+    result['changed'] = True
+
+
+def uninstall_flat(module, binary, name, method):
+    """Remove an existing flatpak."""
+    global result
+    installed_flat_name = _match_installed_flat_name(module, binary, name, method)
+    command = "{0} uninstall --{1} {2}".format(binary, method, installed_flat_name)
+    _flatpak_command(module, module.check_mode, command)
+    result['changed'] = True
+
+
+def flatpak_exists(module, binary, name, method):
+    """Check if the flatpak is installed."""
+    command = "{0} list --{1} --app".format(binary, method)
+    output = _flatpak_command(module, False, command)
+    name = _parse_flatpak_name(name).lower()
+    if name in output.lower():
+        return True
+    return False
+
+
+def _match_installed_flat_name(module, binary, name, method):
+    # This is a difficult function because it seems there
+    # is no naming convention for the flatpakref to what
+    # the installed flatpak will be named.
+    global result
+    command = "{0} list --{1} --app".format(binary, method)
+    output = _flatpak_command(module, False, command)
+    parsed_name = _parse_flatpak_name(name)
+    for row in output.split('\n'):
+        if parsed_name.lower() in row.lower():
+            return row.split()[0]
+
+    result['msg'] = "Flatpak removal failed: Could not match any installed flatpaks to " +\
+        "the name `{0}`. ".format(_parse_flatpak_name(name)) +\
+        "If you used a URL, try using the reverse DNS name of the flatpak"
+    module.fail_json(**result)
+
+
+def _parse_flatpak_name(name):
+    if 'http://' in name or 'https://' in name:
```

Comment:
Since we expect the name to start with 'http://' or 'https://' I would so this:
```python
    if name.startswith('http://') or name.startswith('https://'):
        ...
```